Mathlib/Topology/Order/Monotone.lean

```diff
@@ -246,18 +246,67 @@ theorem Antitone.map_ciInf_of_continuousAt {f : α → β} {g : γ → α} (Cf :
   Monotone.map_ciInf_of_continuousAt (show ContinuousAt (OrderDual.toDual ∘ f) (⨅ i, g i) from Cf)
     Af H
 
-/-- A monotone map has a limit to the left of any point `x`, equal to `sSup (f '' (Iio x))`. -/
-theorem Monotone.tendsto_nhdsWithin_Iio {α β : Type*} [LinearOrder α] [TopologicalSpace α]
+lemma MonotoneOn.tendsto_nhdsWithin_Ioo_left {α β : Type*} [LinearOrder α] [TopologicalSpace α]
+    [OrderTopology α] [ConditionallyCompleteLinearOrder β] [TopologicalSpace β] [OrderTopology β]
+    {f : α → β} {x y : α} (h_nonempty : (Ioo y x).Nonempty) (Mf : MonotoneOn f (Ioo y x))
+    (h_bdd : BddAbove (f '' Ioo y x)) :
+    Tendsto f (𝓝[<] x) (𝓝 (sSup (f '' Ioo y x))) := by
+  refine tendsto_order.2 ⟨fun l hl => ?_, fun m hm => ?_⟩
+  · obtain ⟨z, ⟨yz, zx⟩, lz⟩ : ∃ a : α, a ∈ Ioo y x ∧ l < f a := by
+      simpa only [mem_image, exists_prop, exists_exists_and_eq_and] using
+        exists_lt_of_lt_csSup (h_nonempty.image _) hl
+    refine mem_of_superset (Ioo_mem_nhdsWithin_Iio' zx) fun w hw => ?_
+    exact lz.trans_le <| Mf ⟨yz, zx⟩ ⟨yz.trans_le hw.1.le, hw.2⟩ hw.1.le
+  · rcases h_nonempty with ⟨_, hy, hx⟩
+    refine mem_of_superset (Ioo_mem_nhdsWithin_Iio' (hy.trans hx)) fun w hw => lt_of_le_of_lt ?_ hm
+    exact le_csSup h_bdd (mem_image_of_mem _ hw)
+
+lemma MonotoneOn.tendsto_nhdsWithin_Ioo_right {α β : Type*} [LinearOrder α] [TopologicalSpace α]
+    [OrderTopology α] [ConditionallyCompleteLinearOrder β] [TopologicalSpace β] [OrderTopology β]
+    {f : α → β} {x y : α} (h_nonempty : (Ioo x y).Nonempty) (Mf : MonotoneOn f (Ioo x y))
+    (h_bdd : BddBelow (f '' Ioo x y)) :
+    Tendsto f (𝓝[>] x) (𝓝 (sInf (f '' Ioo x y))) := by
+  refine tendsto_order.2 ⟨fun l hl => ?_, fun m hm => ?_⟩
+  · rcases h_nonempty with ⟨p, hy, hx⟩
+    refine mem_of_superset (Ioo_mem_nhdsWithin_Ioi' (hy.trans hx)) fun w hw => hl.trans_le ?_
+    exact csInf_le h_bdd (mem_image_of_mem _ hw)
+  · obtain ⟨z, ⟨xz, zy⟩, zm⟩ : ∃ a : α, a ∈ Ioo x y ∧ f a < m := by
+      simpa [mem_image, exists_prop, exists_exists_and_eq_and] using
+        exists_lt_of_csInf_lt (h_nonempty.image _) hm
+    refine mem_of_superset (Ioo_mem_nhdsWithin_Ioi' xz) fun w hw => ?_
+    exact (Mf ⟨hw.1, hw.2.trans zy⟩ ⟨xz, zy⟩ hw.2.le).trans_lt zm
```

Comment:
There doesn't happen to be a duality argument lean would accept?

Reply:
In the following lemma I managed to use a duality argument, but here I did not find a way to do it